public/components/alerts/alerts.js

```diff
@@ -1,57 +1,61 @@
+import Kefir from 'kefir';
+import CanStream from 'can-stream';
 import Component from 'can-component';
 import DefineMap from 'can-define/map/';
-import template from './alerts.stache';
+import 'can-define-stream';
+import view from './alerts.stache';
 import './alerts.less';
-import hub from '../../lib/hub';
-import AlertItem from '../../models/alert';
+import hub from '../../lib/event-hub';
+import CID from 'can-cid';
 
-/**
- * TODO: use streams for modifying alerts
- *      - https://github.com/donejs/bitcentive/issues/149
- */
+
+var alertStream = CanStream.toStream(hub, 'alert').map(ev => {
+  return Object.assign({
+    id: CID(ev),
+    kind: 'warning'
+  }, ev);
+});
 
 export const ViewModel = DefineMap.extend({
-  subscription: '*',
-  alerts: {
-    Value: AlertItem.List
-  },
-  addAlert (alert) {
-    this.alerts.push(alert);
-  },
-  removeAlert (alert) {
-    const idx = this.alerts.indexOf(alert);
-    if (idx !== -1) {
-      this.alerts.splice(idx, 1);
+  autoHideStream: {
+    value() {
+      return alertStream.flatMap(alert => {
+        if (alert.displayInterval > 0) {
+          return Kefir.later(alert.displayInterval, {
+            type: 'remove',
+            id: alert.id
+          });
+        } 
+        return Kefir.constant({ type: 'no-op' });
```

Comment:
I couldn't figure out a way around this as flatMap has to return _something_. I can filter out the 'no-op' events if you want - akin to `[].map(...).filter(...)`.